### PR TITLE
[Impeller] Added a switch to turn on vulkan

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/128286

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gaaclarke** [comment]
Dan is not available today so I added @jason-simmons

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -41,6 +41,7 @@ std::unique_ptr<AndroidSurface> AndroidSurfaceFactoryImpl::CreateSurface() {
     case AndroidRenderingAPI::kSoftware:
       return std::make_unique<AndroidSurfaceSoftware>();
     case AndroidRenderingAPI::kOpenGLES:
+      FML_DLOG(ERROR) << "using opengl";
```
Comment (jason-simmons) [inline shell/platform/android/platform_view_android.cc:44]: Remove this and change the "Using the Impeller rendering backend" log in `android_surface_gl_impeller.cc` to mention OpenGL ES

File: shell/platform/android/platform_view_android.cc [hunk 2]
```diff
@@ -51,16 +52,10 @@ std::unique_ptr<AndroidSurface> AndroidSurfaceFactoryImpl::CreateSurface() {
       }
     case AndroidRenderingAPI::kVulkan:
       FML_DCHECK(enable_impeller_);
-      // TODO(kaushikiska@): Enable this after wiring a preference for Vulkan
-      // backend.
-#if false
-    return std::make_unique<AndroidSurfaceVulkanImpeller>(
+      FML_DLOG(ERROR) << "using vulkan";
```
Comment (jason-simmons) [inline shell/platform/android/platform_view_android.cc:55]: Remove this and add a "Using the Impeller Vulkan rendering backend" log to `android_surface_vulkan_impeller.cc` similar to the one for OpenGL ES

File: shell/common/switches.h [hunk 3]
```diff
@@ -261,6 +261,10 @@ DEF_SWITCH(EnableImpeller,
            "enable-impeller",
            "Enable the Impeller renderer on supported platforms. Ignored if "
            "Impeller is not supported on the platform.")
+DEF_SWITCH(ImpellerBackend,
+           "impeller-backend",
+           "Hardcodes a selection of which impeller backend to use. (ex "
```
Comment (jason-simmons) [inline shell/common/switches.h:266]: Change the description (and the comment in `settings.h`) to reflect that this is modifiable and not hardcoded
(e.g. "Requests a particular Impeller backend on platforms that support multiple backends")

Comment (gaaclarke) [reply shell/common/switches.h:266]: done

**Comment by gaaclarke** [comment]
I've pushed 2 more commits to this branch and it doesn't seem to be showing up in the pr: https://github.com/gaaclarke/engine/commits/vulkan-switch

**Comment by gaaclarke** [comment]
This seems to be a known github issue as reported by https://www.githubstatus.com/

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**